src/amqp_management.c

```diff
@@ -343,18 +355,24 @@ static void on_message_send_complete(void* context, MESSAGE_SEND_RESULT send_res
     }
     else
     {
+        /* Codes_SRS_AMQP_MANAGEMENT_01_168: [ - `context` shall be used as a LIST_ITEM_HANDLE containing the pending operation. ]*/
+        LIST_ITEM_HANDLE pending_operation_list_item_handle = (LIST_ITEM_HANDLE)context;
+
+        /* Codes_SRS_AMQP_MANAGEMENT_01_169: [ - `on_message_send_complete` shall obtain the pending operation by calling `singlylinkedlist_item_get_value`. ]*/
+        OPERATION_MESSAGE_INSTANCE* pending_operation_message = (OPERATION_MESSAGE_INSTANCE*)singlylinkedlist_item_get_value(pending_operation_list_item_handle);
```

Comment:
are you sure there will always be one item on this list?

Reply:
It should. It is the entry that was created when messagesender_send_async was called. If it is not there, a major memory issue has happened outside the SDK code.

Reply:
No, I mean more then 1.